resource_alks_iamrole.go

```diff
@@ -195,8 +195,17 @@ func resourceAlksIamRoleRead(ctx context.Context, d *schema.ResourceData, meta i
 	foundRole, err := client.GetIamRole(d.Id())
 
 	if err != nil {
+		// If 404 Role not found error, an error and a role with Exists field set to false will come back from alks-go
+		// We will log ther error and set id to "" and return nil, letting terraform decide how to handle role not found.
+		if foundRole != nil {
+			if foundRole.Exists != true {
+				log.Printf("[Error] %#v", err)
+				d.SetId("")
```

Comment:
Is the # mean to be in the format string?

Reply:
Oh yeah good call.  I sometimes use that when I'm unsure of the type because I've had it work before when other stuff errored out.  I've changed it to %s and tested it.